lib/block_view.go

```diff
@@ -4058,35 +4058,52 @@ func (bav *UtxoView) ConnectBlock(
 	for txIndex, txn := range desoBlock.Txns {
 		txHash := txHashes[txIndex]
 
-		// ConnectTransaction validates all of the transactions in the block and
-		// is responsible for verifying signatures.
-		//
-		// TODO: We currently don't check that the min transaction fee is satisfied when
-		// connecting blocks. We skip this check because computing the transaction's size
-		// would slow down block processing significantly. We should figure out a way to
-		// enforce this check in the future, but for now the only attack vector is one in
-		// which a miner is trying to spam the network, which should generally never happen.
-		utxoOpsForTxn, totalInput, totalOutput, currentFees, err := bav.ConnectTransaction(txn, txHash, 0, uint32(blockHeader.Height), blockHeader.TstampNanoSecs, verifySignatures, false)
-		_, _ = totalInput, totalOutput // A bit surprising we don't use these
 		// After the PoS cutover, we need to check if the transaction is a failing transaction.
 		txnConnects := blockHeight < uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) ||
 			(txIndex == 0 && txn.TxnMeta.GetTxnType() == TxnTypeBlockReward) ||
 			desoBlock.TxnConnectStatusByIndex.Get(txIndex-1)
```

Comment:
What happens if `txIndex = 0 && txn.TxnMeta.GetTxnType() != TxnTypeBlockReward`? Won't desoBlock.TxnConnectStatusByIndex.Get(-1) panic?

Reply:
On line 4038 in block_view.go, we verify that the first transaction is a block reward transaction.

Reply:
Ah, good, good!